src/Catalog/Monitoring/Model/PackageTimestampMetadata.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Protocol.Core.Types;
+using Newtonsoft.Json.Linq;
+using NuGet.Services.Metadata.Catalog.Helpers;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Providers
+{
+    /// <summary>
+    /// Represents the timestamp metadata for a single package in a package source.
+    /// </summary>
+    public class PackageTimestampMetadata : INuGetResource
+    {
+        public PackageTimestampMetadata()
+        {
+            Exists = false;
+            Created = null;
+            LastEdited = null;
+            Deleted = null;
+        }
+
+        public PackageTimestampMetadata(DateTime? deleted)
+        {
+            Exists = false;
+            Created = null;
+            LastEdited = null;
+            Deleted = deleted;
+        }
+
+        public PackageTimestampMetadata(DateTime created, DateTime lastEdited)
+        {
+            Exists = true;
+            Created = created;
+            LastEdited = lastEdited;
+            Deleted = null;
+        }
+
+        public static async Task<PackageTimestampMetadata> FromCatalogEntry(CollectorHttpClient client,
+            CatalogIndexEntry catalogEntry)
+        {
+            var catalogPage = await client.GetJObjectAsync(catalogEntry.Uri);
+
+            if (catalogEntry.IsDelete())
+            {
+                // On the catalog page for a delete, the published value is the timestamp the package was deleted from the audit records.
+                var deleted = catalogPage.GetValue("published").Value<DateTimeOffset>();
```

Comment:
This can throw if GetValue returns null, and it will throw with the very helpful exception: "System.ArgumentNullException: 'Value cannot be null.'"

If we are going to use an exception log to try to diagnose issues, we may want a more descriptive exception to be thrown here and in other places we have this pattern.